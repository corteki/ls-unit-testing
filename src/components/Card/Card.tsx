import React, { FC } from 'react';
import { CardProps } from './CardProps';

export const Card: FC<CardProps> = ({title, body, footer, children}) => (
  <article>
    <h2>{title}</h2>
    <div>
      <p>{body}</p>
      <small>{footer}</small>
      {children}
    </div>
  </article>
)