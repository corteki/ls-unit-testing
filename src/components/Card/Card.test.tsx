import React from "react";
import renderer from "react-test-renderer";
import { Card } from "./Card";
import { MOCK_TITLE, MOCK_BODY, MOCK_FOOTER } from "./fixtures";

describe('<Card/>', () => {
  it('should display a Card with a title, body, footer and addionally some children', () => {
    const tree = renderer
    .create(
      <Card
      title={MOCK_TITLE}
      body={MOCK_BODY}
      footer={MOCK_FOOTER}>
        <p>child</p>
      </Card>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  })
})