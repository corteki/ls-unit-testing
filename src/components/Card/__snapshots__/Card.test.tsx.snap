// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should display a Card with a title, body, footer and addionally some children 1`] = `
<article>
  <h2>
    MockTitle
  </h2>
  <div>
    <p>
      MockBody
    </p>
    <small>
      MockFooter
    </small>
    <p>
      child
    </p>
  </div>
</article>
`;
